rich/text.py

```diff
@@ -141,15 +140,16 @@ def __init__(
         tab_size: Optional[int] = 8,
         spans: Optional[List[Span]] = None,
     ) -> None:
-        self._text = [strip_control_codes(text)]
+        sanitized_text = make_control_codes_readable(text)
+        self._text = [sanitized_text]
         self.style = style
         self.justify: Optional["JustifyMethod"] = justify
         self.overflow: Optional["OverflowMethod"] = overflow
         self.no_wrap = no_wrap
         self.end = end
         self.tab_size = tab_size
         self._spans: List[Span] = spans or []
-        self._length: int = len(text)
+        self._length: int = len(sanitized_text)
```

Comment:
that's the bugfix itself! :slightly_smiling_face: i.e. we should use the length of the sanitised text, rather than the one of the text we received as an argument